code/game/gamemodes/steal_items.dm

```diff
@@ -212,16 +212,18 @@
 		var/found_amount = 0
 		for(var/obj/I in all_items) //Check for items
 			if(istype(I, typepath))
+			
 				//Stealing the cheap autoinjector doesn't count
 				if(istype(I, /obj/item/weapon/reagent_containers/hypospray/autoinjector))
 					continue
+
 				if(istype(I,/obj/item/device/aicard))
 					var/obj/item/device/aicard/C = I
 					if(!C.contents.len)
 						continue //Stealing a card with no contents doesn't count
 					var/is_at_least_one_alive = 0
 					for(var/mob/living/silicon/ai/A in C)
-						if(A.stat != DEAD)
+						if(A.stat != 2)
```

Comment:
The reason the AI card objective isn't working is because the block of code listed here is never called for it.

AI card is `/datum/theft_objective/traitor/ai` and this code block is for types of `/datum/theft_objective/number`.

The best way to fix this is to take the entire if(istype) block from here and instead add in  `/datum/theft_objective/traitor/ai/check_completion()`.

```
/datum/theft_objective/traitor/ai/check_completion(var/datum/mind/owner)
 	if(!C.contents.len)
  		continue //Stealing a card with no contents doesn't count
  	var/is_at_least_one_alive = 0
  	for(var/mob/living/silicon/ai/A in C)
 		if(A.stat != DEAD)
			is_at_least_one_alive++
 	if(is_at_least_one_alive)
  		return ..()
```